### PR TITLE
Optimizations for `script.toAddress()`

### PR DESCRIPTION
**Pull request by braydonf**
Several optimizations for `script.toAddress()` for around a 4x speed improvement.

### CODE REVIEW

File: lib/address.js [hunk 1]
```diff
@@ -505,3 +504,5 @@ Address.prototype.inspect = function() {
 };
 
 module.exports = Address;
+
+var Script = require('./script');
```
Comment (maraoz) [inline lib/address.js]: why this change?

**Comment by maraoz** [comment]
LGTM

File: lib/address.js (hunk 1 above)
Comment (braydonf) [inline lib/address.js]: Resolves the circular dependency between `address.js` and `script.js` without needing to have several `address.js` require statements in functions in `script.js` that slowed each function down each time it was invoked. Using a late definition here is faster.

**Comment by pnagurny** [comment]
LGTM
